scripts/west_commands/runners/blackmagicprobe.py

```diff
@@ -150,16 +151,25 @@ def do_add_parser(cls, parser):
                             help='Assert SRST during connect? (default: no)')
 
     def bmp_flash(self, command, **kwargs):
-        if self.elf_file is None:
-            raise ValueError('Cannot debug; elf file is missing')
+        # if hex file is not present or not signed, use the elf file
+        if self.hex_file is None:
```

Comment:
could you swap the condition? I think it's easier to follow the logic if the case using `hex_file` is under `if self.hex_file:` rather than the nagated one